tutor/plugins/openedx.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+import os
+import re
+import typing as t
+
+from tutor import bindmount
+from tutor import hooks
+
+
+@hooks.Filters.APP_PUBLIC_HOSTS.add()
+def _edx_platform_public_hosts(
+    hosts: list[str], context_name: t.Literal["local", "dev"]
+) -> list[str]:
+    if context_name == "dev":
+        hosts += ["{{ LMS_HOST }}:8000", "{{ CMS_HOST }}:8001"]
+    else:
+        hosts += ["{{ LMS_HOST }}", "{{ CMS_HOST }}"]
+    return hosts
+
+
+@hooks.Filters.IMAGES_BUILD_MOUNTS.add()
+def _mount_edx_platform_build(
+    volumes: list[tuple[str, str]], path: str
+) -> list[tuple[str, str]]:
+    """
+    Automatically add an edx-platform repo from the host to the build context whenever
+    it is added to the `MOUNTS` setting.
+    """
+    if os.path.basename(path) == "edx-platform":
+        volumes += [
+            ("openedx", "edx-platform"),
+            ("openedx-dev", "edx-platform"),
+        ]
+    return volumes
+
+
+@hooks.Filters.COMPOSE_MOUNTS.add()
+def _mount_edx_platform_compose(
+    volumes: list[tuple[str, str]], name: str
+) -> list[tuple[str, str]]:
+    """
+    When mounting edx-platform with `tutor mounts add /path/to/edx-platform`,
+    bind-mount the host repo in the lms/cms containers.
+    """
+    if name == "edx-platform":
+        path = "/openedx/edx-platform"
+        volumes += [
+            ("lms", path),
+            ("cms", path),
+            ("lms-worker", path),
+            ("cms-worker", path),
+            ("lms-job", path),
+            ("cms-job", path),
+        ]
+    return volumes
+
+
+# Auto-magically bind-mount xblock directories and some common dependencies.
+hooks.Filters.EDX_PLATFORM_PYTHON_PACKAGES.add_items(
+    [
+        r".*[xX][bB]lock.*",
+        "edx-enterprise",
+        "edx-ora2",
+        "edx-search",
```

Comment:
```suggestion
    [
        r".*[xX][bB]lock.*",
        "edx-enterprise",
        "edx-ora2",
        "edx-search",
        "platform-plugin-.*",
```
The `platform-plugin-` prefix is a convention that some teams used when I was at edX. I think it's nice, and if you don't hate it, then I'd request you add it to the list :)